### PR TITLE
Split up drawing functions for covered track elements

### PR DESCRIPTION
**Pull request by Gymnasiast**
Some time ago, @IntelOrca and I came up with a way to simplify the track drawing functions a bit. Rather than having the track paint functions figure out covered tiles and whether a part is inverted, this would be split into separate track drawing functions.

This is part 1 of that. It split out drawing covered sections. These are now passed to a track drawer in normalised form. E.g. a covered flat dinghy slide piece is now passed to the DinghySlideCovered drawer as a regular flat piece. Similarly, a "covered" flat water coaster piece (i.e. submerged track) is now passed to the SplashBoats drawer as a regular flat piece directly.

Once the appropriate track element refactor is in, this will mean that _any_ track element can now be covered. In practical terms, that would make stuff like this possible:
- Allowing covered lift hills on the Dinghy Slide
- Building an on-ride photo on the water section of the Water Coaster
- Allowing switching between two-rail and three-rail Intamin/Mack/Gerstlauer-like track

### CODE REVIEW

File: src/openrct2/ride/TrackPaint.cpp [hunk 1]
```diff
@@ -2254,7 +2254,16 @@ void PaintTrack(PaintSession& session, Direction direction, int32_t height, cons
...
-        TRACK_PAINT_FUNCTION_GETTER paintFunctionGetter = rtd.TrackPaintFunction;
+        auto paintFunctions = rtd.TrackPaintFunctions;
+        TRACK_PAINT_FUNCTION_GETTER paintFunctionGetter;
+        if (TrackElementIsCovered(trackType))
+        {
+            trackType = UncoverTrackElement(trackType);
+            paintFunctionGetter = paintFunctions.Covered;
+        }
+        else
+            paintFunctionGetter = paintFunctions.Regular;
```
Comment (IntelOrca) [inline src/openrct2/ride/TrackPaint.cpp:2265]: Usually we brace niether or both blocks, not mixed.

**Comment by karst** [comment]
I'll give it a test. I will give my review in a bit.

Question, will this allow for multiple entries, not just two?
The track switching for the intamin track is going to have three types of spines to switch between.

**Comment by IntelOrca** [comment]
Yes, at present this is just a refactor exercise so we have discrete track paint modules for covered and non-covered.

**Review by karst — APPROVED** [review]
My tests:
- Before this PR: When changing between track types covered pieces will be empty
- With this PR: When changing between track types covered pieces will resemble their non-covered counterpart.

Currently, when switching the ride type it remembers what track type was used when swapping back, but I thought one of the issues would become when using the track type switches that when doing a ride type change it will not remember those different types? Or is this fixed by giving a specific number to the used track type? How does this function?

Everything seems to be working as intended.

**Comment by karst** [comment]
Technically speaking you're making something that used to be invisible, visible.... Maybe adding a conversion code could be useful here, but it would add a bunch of bloat for something that's probably barely or never used.

**Comment by Gymnasiast** [comment]
I pushed an updated version that should address both the line note as well as the visibility issue.

**Review by karst — APPROVED** [review]
Everything seems to be in working order (same behaviour as before this refactor), and as a bonus it will also fix  #21074.

**Pull request merged**